code/mob/living/critter/ai/maneater.dm

```diff
@@ -0,0 +1,49 @@
+
+/datum/aiHolder/maneater
+	New()
+		..()
+		default_task = get_instance(/datum/aiTask/prioritizer/critter/maneater, list(src))
+
+
+/datum/aiTask/prioritizer/critter/maneater
+	name = "Maneater priorisation"
+
+/datum/aiTask/prioritizer/critter/maneater/New()
+	..()
+	transition_tasks += holder.get_instance(/datum/aiTask/timed/wander/critter/aggressive, list(holder, src))
+	transition_tasks += holder.get_instance(/datum/aiTask/sequence/goalbased/critter/manhunter_hunting, list(holder, src))
+	transition_tasks += holder.get_instance(/datum/aiTask/sequence/goalbased/critter/eat, list(holder, src))
+
+/datum/aiTask/prioritizer/critter/maneater/on_reset()
+	..()
+	holder.stop_move()
+
+/datum/aiTask/sequence/goalbased/critter/manhunter_hunting
+	name = "attacking"
+	weight = 10 // this behaviour gets a high priority
+	ai_turbo = TRUE
+	max_dist = 7
+
+/datum/aiTask/sequence/goalbased/critter/manhunter_hunting/get_targets()
+	var/mob/living/critter/task_owner = holder.owner
+	return task_owner.seek_target(src.max_dist)
+
+/datum/aiTask/sequence/goalbased/critter/manhunter_hunting/precondition()
+	var/mob/living/critter/task_owner = holder.owner
+	return task_owner.can_critter_attack()
+
+/datum/aiTask/sequence/goalbased/critter/manhunter_hunting/score_target(atom/target)
+	. = 0
+	if(target)
```

Comment:
You can ditch this check, `istype()` returns false on `null` values too, so it's redundant.

Reply:
Makes thanks, will change that directly, thanks!